tools/gen_selectedops.py

```diff
@@ -1,80 +1,103 @@
-import os
+#!/usr/bin/env python3
+
+import argparse
+import pathlib
 import sys
 
 OPMAP_TO_CMAKE_FLAGS = {
-    'BlingFireSentenceBreaker': 'OCOS_ENABLE_BLINGFIRE',
-    'GPT2Tokenizer': 'OCOS_ENABLE_GPT2_TOKENIZER',
-    'WordpieceTokenizer': 'OCOS_ENABLE_WORDPIECE_TOKENIZER',
-    'StringConcat': 'OCOS_ENABLE_TF_STRING',
-    'StringECMARegexReplace': 'OCOS_ENABLE_TF_STRING',
-    'StringECMARegexSplitWithOffsets': 'OCOS_ENABLE_TF_STRING',
-    'StringEqual': 'OCOS_ENABLE_TF_STRING',
-    'StringToHashBucket': 'OCOS_ENABLE_TF_STRING',
-    'StringToHashBucketFast': 'OCOS_ENABLE_TF_STRING',
-    'StringJoin': 'OCOS_ENABLE_TF_STRING',
-    'StringLength': 'OCOS_ENABLE_TF_STRING',
-    'StringLower': 'OCOS_ENABLE_TF_STRING',
-    'StringRegexReplace': 'OCOS_ENABLE_RE2_REGEX',
-    'StringRegexSplitWithOffsets': 'OCOS_ENABLE_RE2_REGEX',
-    'StringSplit': 'OCOS_ENABLE_TF_STRING',
-    'StringToVector': 'OCOS_ENABLE_TF_STRING',
-    'StringUpper': 'OCOS_ENABLE_TF_STRING',
-    'SegmentExtraction': 'OCOS_ENABLE_MATH',
-    'StringMapping': 'OCOS_ENABLE_TF_STRING',
-    'VectorToString': 'OCOS_ENABLE_TF_STRING',
-    'MaskedFill': 'OCOS_ENABLE_TF_STRING',
-    'BertTokenizer': 'OCOS_ENABLE_BERT_TOKENIZER',
-    'BasicTokenizer': 'OCOS_ENABLE_BERT_TOKENIZER',
-    'BertTokenizerDecoder': 'OCOS_ENABLE_BERT_TOKENIZER',
-    'SentencepieceTokenizer': 'OCOS_ENABLE_SPM_TOKENIZER',
-    'ImageDecode': 'OCOS_ENABLE_VISION',
-    'ImageEncode': 'OCOS_ENABLE_VISION',
+    "BlingFireSentenceBreaker": "OCOS_ENABLE_BLINGFIRE",
+    "GPT2Tokenizer": "OCOS_ENABLE_GPT2_TOKENIZER",
+    "WordpieceTokenizer": "OCOS_ENABLE_WORDPIECE_TOKENIZER",
+    "StringConcat": "OCOS_ENABLE_TF_STRING",
+    "StringECMARegexReplace": "OCOS_ENABLE_TF_STRING",
+    "StringECMARegexSplitWithOffsets": "OCOS_ENABLE_TF_STRING",
+    "StringEqual": "OCOS_ENABLE_TF_STRING",
+    "StringToHashBucket": "OCOS_ENABLE_TF_STRING",
+    "StringToHashBucketFast": "OCOS_ENABLE_TF_STRING",
+    "StringJoin": "OCOS_ENABLE_TF_STRING",
+    "StringLength": "OCOS_ENABLE_TF_STRING",
+    "StringLower": "OCOS_ENABLE_TF_STRING",
+    "StringRegexReplace": "OCOS_ENABLE_RE2_REGEX",
+    "StringRegexSplitWithOffsets": "OCOS_ENABLE_RE2_REGEX",
+    "StringSplit": "OCOS_ENABLE_TF_STRING",
+    "StringToVector": "OCOS_ENABLE_TF_STRING",
+    "StringUpper": "OCOS_ENABLE_TF_STRING",
+    "SegmentExtraction": "OCOS_ENABLE_MATH",
+    "StringMapping": "OCOS_ENABLE_TF_STRING",
+    "VectorToString": "OCOS_ENABLE_TF_STRING",
+    "MaskedFill": "OCOS_ENABLE_TF_STRING",
+    "BertTokenizer": "OCOS_ENABLE_BERT_TOKENIZER",
+    "BasicTokenizer": "OCOS_ENABLE_BERT_TOKENIZER",
+    "BertTokenizerDecoder": "OCOS_ENABLE_BERT_TOKENIZER",
+    "SentencepieceTokenizer": "OCOS_ENABLE_SPM_TOKENIZER",
+    "DecodeImage": "OCOS_ENABLE_VISION",
```

Comment:
it may not directly relate to this PR, but I want to have discussion about the Op name. In most case, there *Tokenizer, or *decoder. so it would be consistent to name them as ImageDecoder and ImageEncoder?

Reply:
I named them this way to avoid a clash with the opencv ops as there's an `ImageDecoder` there. 

I would like to separate the operators that are 1:1 with opencv functions and behavior from operators that may or may not use opencv as a dependency. e.g. we may directly use libpng and libjpeg-turbo for the pre/post processing to reduce binary size, but currently are using opencv due to the timeline. 

The vision operators are aimed at providing the required decode/encode for mobile scenarios rather than being a 1:1 with opencv. e.g. I need to support RGB input for web (basically a no-op decode), and also want to make the output layout configurable so it could be RGB or BGR instead of always being the (slightly odd) opencv BGR layout. 

If we rename the opencv operators I'm fine with changing general purpose image operators to use 'Image' as the prefix. Not sure what a good naming scheme would be though. Use a different domain for the opencv ops and the function name (but function name is all lowercase so that's slightly inconsistent with operator names in general)?